lib/adapters/REST/endpoints/app-upload.ts

```diff
@@ -0,0 +1,51 @@
+import type { AxiosInstance } from 'contentful-sdk-core'
+import { Stream } from 'stream'
+import * as raw from './raw'
+import { GetAppUploadParams, GetOrganizationParams } from '../../../common-types'
+import { RestEndpoint } from '../types'
+import { AppUploadProps } from '../../../entities/app-upload'
+import { getUploadHttpClient } from '../../../upload-http-client'
+
+const getBaseUrl = (params: GetOrganizationParams) =>
+  `/organizations/${params.organizationId}/app_uploads`
+
+const getAppUploadUrl = (params: GetAppUploadParams) =>
+  `${getBaseUrl(params)}/${params.appUploadId}`
+
+export const get: RestEndpoint<'AppUpload', 'get'> = (
+  http: AxiosInstance,
+  params: GetAppUploadParams
+) => {
+  const httpUpload = getUploadHttpClient(http)
+
+  return raw.get<AppUploadProps>(httpUpload, getAppUploadUrl(params))
+}
+
+export const del: RestEndpoint<'AppUpload', 'delete'> = (
+  http: AxiosInstance,
+  params: GetAppUploadParams
+) => {
+  const httpUpload = getUploadHttpClient(http)
+
+  return raw.del<void>(httpUpload, getAppUploadUrl(params))
+}
+
+export const create: RestEndpoint<'AppUpload', 'create'> = async (
+  http: AxiosInstance,
+  params: GetOrganizationParams,
+  payload: { file: string | ArrayBuffer | Stream }
+) => {
+  const httpUpload = getUploadHttpClient(http)
+
+  const { file } = payload
+
+  if (!file) {
+    return Promise.reject(new Error('Unable to locate a file to upload.'))
+  }
```

Comment:
How could this happen? And do we need an extra check for that?

We also don't check whether the organizationId is an empty string or undefined.

Reply:
Probably not. To be honest I stuck to the implementation we already use for the `Upload` entity.  In my opinion we can remove this and add a check for the `organizationId`.

Reply:
Okay let's leave it as it is for now.